// @fern-api/phyllo/core/schemas/builders/object/object.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var object_exports = {};
__export(object_exports, {
  getObjectUtils: () => getObjectUtils,
  object: () => object
});
module.exports = __toCommonJS(object_exports);
var import_entries = require("../../utils/entries");
var import_object_like = require("../object-like");
var import_schema_utils = require("../schema-utils");
var import_property = require("./property");
function object(schemas) {
  const baseSchema = {
    ...import_object_like.OBJECT_LIKE_BRAND,
    parse: (raw, { skipUnknownKeysOnParse = false } = {}) => {
      const rawKeyToProperty = {};
      for (const [parsedKey, schemaOrObjectProperty] of (0, import_entries.entries)(schemas)) {
        const rawKey = (0, import_property.isProperty)(schemaOrObjectProperty) ? schemaOrObjectProperty.rawKey : parsedKey;
        const property = {
          rawKey,
          parsedKey,
          valueSchema: (0, import_property.isProperty)(schemaOrObjectProperty) ? schemaOrObjectProperty.valueSchema : schemaOrObjectProperty
        };
        rawKeyToProperty[rawKey] = property;
      }
      const parsed = {};
      for (const [rawKey, rawPropertyValue] of Object.entries(raw)) {
        const property = rawKeyToProperty[rawKey];
        if (property != null) {
          const value = property.valueSchema.parse(rawPropertyValue);
          parsed[property.parsedKey] = value;
        } else if (!skipUnknownKeysOnParse && rawPropertyValue != null) {
          parsed[rawKey] = rawPropertyValue;
        }
      }
      return parsed;
    },
    json: (parsed, { includeUnknownKeysOnJson = false } = {}) => {
      const raw = {};
      for (const [parsedKey, parsedPropertyValue] of (0, import_entries.entries)(parsed)) {
        const schemaOrObjectProperty = schemas[parsedKey];
        if (schemaOrObjectProperty != null) {
          if ((0, import_property.isProperty)(schemaOrObjectProperty)) {
            const value = schemaOrObjectProperty.valueSchema.json(parsedPropertyValue);
            raw[schemaOrObjectProperty.rawKey] = value;
          } else {
            const value = schemaOrObjectProperty.json(parsedPropertyValue);
            raw[parsedKey] = value;
          }
        } else if (includeUnknownKeysOnJson && parsedPropertyValue != null) {
          raw[parsedKey] = parsedPropertyValue;
        }
      }
      return raw;
    }
  };
  return {
    ...baseSchema,
    ...(0, import_schema_utils.getSchemaUtils)(baseSchema),
    ...(0, import_object_like.getObjectLikeUtils)(baseSchema),
    ...getObjectUtils(baseSchema)
  };
}
function getObjectUtils(schema) {
  return {
    extend: (extension) => {
      const baseSchema = {
        ...import_object_like.OBJECT_LIKE_BRAND,
        parse: (raw, opts) => ({
          ...schema.parse(raw, opts),
          ...extension.parse(raw, opts)
        }),
        json: (parsed, opts) => ({
          ...schema.json(parsed, opts),
          ...extension.json(parsed, opts)
        })
      };
      return {
        ...baseSchema,
        ...(0, import_schema_utils.getSchemaUtils)(baseSchema),
        ...(0, import_object_like.getObjectLikeUtils)(baseSchema),
        ...getObjectUtils(baseSchema)
      };
    }
  };
}
