// @fern-api/phyllo/resources/connect/client/Client.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var Client_exports = {};
__export(Client_exports, {
  Client: () => Client
});
module.exports = __toCommonJS(Client_exports);
var environments = __toESM(require("../../../environments"));
var core = __toESM(require("../../../core"));
var import_url_join = __toESM(require("url-join"));
var serializers = __toESM(require("../../../serialization"));
class Client {
  constructor(options) {
    this.options = options;
  }
  async getWorkPlatform(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, `/v1/work-platforms/${request.id}`),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      }
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.WorkPlatform.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getAllWorkPlatforms(request) {
    const queryParameters = new URLSearchParams();
    if (request.limit != null) {
      queryParameters.append("limit", request.limit.toString());
    }
    if (request.name != null) {
      queryParameters.append("name", request.name);
    }
    if (request.offset != null) {
      queryParameters.append("offset", request.offset.toString());
    }
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, "/v1/work-platforms"),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      },
      queryParameters
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.AllWorkPlatformsResponse.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getUser(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, `/v1/users/${request.id}`),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      }
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.User.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getAllUsers(request) {
    const queryParameters = new URLSearchParams();
    if (request.limit != null) {
      queryParameters.append("limit", request.limit.toString());
    }
    if (request.offset != null) {
      queryParameters.append("offset", request.offset.toString());
    }
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, "/v1/users"),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      },
      queryParameters
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.AllUsersResponse.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async createUser(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, "/v1/users"),
      method: "POST",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      },
      body: serializers.connect.CreateUserRequest.json(request)
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.User.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getUserByExernalId(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, `/v1/users/external_id/${request.externalId}`),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      }
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.User.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async createSdkToken(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, "/v1/sdk-tokens"),
      method: "POST",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      },
      body: serializers.connect.SDKTokenRequest.json(request)
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.SDKTokenResponse.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getAccount(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, `/v1/accounts/${request.id}`),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      }
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.Account.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async getAllAccounts(request) {
    const queryParameters = new URLSearchParams();
    if (request.limit != null) {
      queryParameters.append("limit", request.limit.toString());
    }
    if (request.offset != null) {
      queryParameters.append("offset", request.offset.toString());
    }
    if (request.userId != null) {
      queryParameters.append("user_id", request.userId);
    }
    if (request.workPlatformId != null) {
      queryParameters.append("work_platform_id", request.workPlatformId);
    }
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, "/v1/accounts"),
      method: "GET",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      },
      queryParameters
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.AllAccountsResponse.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
  async disconnectAccount(request) {
    const response = await core.fetcher({
      url: (0, import_url_join.default)(this.options.environment ?? environments.Environment.Production, `/v1/accounts/${request.id}/disconnect`),
      method: "POST",
      headers: {
        Authorization: core.BasicAuth.toAuthorizationHeader(await core.Supplier.get(this.options.auth?.credentials))
      }
    });
    if (response.ok) {
      return {
        ok: true,
        body: serializers.connect.DisconnectAccountResponse.parse(response.body)
      };
    }
    return {
      ok: false,
      error: {
        errorName: void 0,
        content: response.error,
        _visit: (visitor) => visitor._other(response.error)
      }
    };
  }
}
