// @fern-api/phyllo/resources/connect/client/index.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var client_exports = {};
__export(client_exports, {
  createSdkToken: () => createSdkToken,
  createUser: () => createUser,
  disconnectAccount: () => disconnectAccount,
  getAccount: () => getAccount,
  getAllAccounts: () => getAllAccounts,
  getAllUsers: () => getAllUsers,
  getAllWorkPlatforms: () => getAllWorkPlatforms,
  getUser: () => getUser,
  getUserByExernalId: () => getUserByExernalId,
  getWorkPlatform: () => getWorkPlatform
});
module.exports = __toCommonJS(client_exports);
var getWorkPlatform = __toESM(require("./getWorkPlatform"));
var getAllWorkPlatforms = __toESM(require("./getAllWorkPlatforms"));
var getUser = __toESM(require("./getUser"));
var getAllUsers = __toESM(require("./getAllUsers"));
var createUser = __toESM(require("./createUser"));
var getUserByExernalId = __toESM(require("./getUserByExernalId"));
var createSdkToken = __toESM(require("./createSdkToken"));
var getAccount = __toESM(require("./getAccount"));
var getAllAccounts = __toESM(require("./getAllAccounts"));
var disconnectAccount = __toESM(require("./disconnectAccount"));
