// @fern-api/phyllo/serialization/connect/types/Account.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var Account_exports = {};
__export(Account_exports, {
  Account: () => Account
});
module.exports = __toCommonJS(Account_exports);
var core = __toESM(require("../../../core"));
var serializers = __toESM(require("../.."));
const Account = core.schemas.object({
  id: core.schemas.lazy(() => serializers.connect.AccountId),
  createdAt: core.schemas.property("created_at", core.schemas.date()),
  updatedAt: core.schemas.property("updated_at", core.schemas.date()),
  user: core.schemas.lazyObject(() => serializers.connect.UserAttribute),
  workPlatform: core.schemas.property("work_platform", core.schemas.lazyObject(() => serializers.connect.WorkPlatformAttribute)),
  platformUsername: core.schemas.property("platform_username", core.schemas.string().optional()),
  profilePicUrl: core.schemas.property("profile_pic_url", core.schemas.string().optional()),
  status: core.schemas.lazy(() => serializers.connect.AccountStatus),
  platformProfileName: core.schemas.property("platform_profile_name", core.schemas.string().optional()),
  platformProfileId: core.schemas.property("platform_profile_id", core.schemas.string().optional()),
  platformProfilePublishedAt: core.schemas.property("platform_profile_published_at", core.schemas.date().optional()),
  data: core.schemas.lazyObject(() => serializers.connect.AccountData)
});
