// @fern-api/phyllo/serialization/connect/types/WorkPlatform.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var WorkPlatform_exports = {};
__export(WorkPlatform_exports, {
  WorkPlatform: () => WorkPlatform
});
module.exports = __toCommonJS(WorkPlatform_exports);
var core = __toESM(require("../../../core"));
var serializers = __toESM(require("../.."));
const WorkPlatform = core.schemas.object({
  createdAt: core.schemas.property("created_at", core.schemas.date()),
  updatedAt: core.schemas.property("updated_at", core.schemas.date()),
  category: core.schemas.lazy(() => serializers.connect.WorkPlatformCategory),
  status: core.schemas.lazy(() => serializers.connect.Status),
  url: core.schemas.string(),
  products: core.schemas.lazyObject(() => serializers.connect.Products)
}).extend(core.schemas.lazyObject(() => serializers.connect.WorkPlatformAttribute));
