// @fern-api/phyllo/resources/connect/types/DataStatus.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var DataStatus_exports = {};
__export(DataStatus_exports, {
  DataStatus: () => DataStatus
});
module.exports = __toCommonJS(DataStatus_exports);
const _Synced = {
  value: "SYNCED",
  visit: (visitor) => visitor.synced()
};
const _InProgress = {
  value: "IN_PROGRESS",
  visit: (visitor) => visitor.inProgress()
};
const _NotSupported = {
  value: "NOT_SUPPORTED",
  visit: (visitor) => visitor.notSupported()
};
const _NotSynced = {
  value: "NOT_SYNCED",
  visit: (visitor) => visitor.notSynced()
};
const _PartialSynced = {
  value: "PARTIAL_SYNCED",
  visit: (visitor) => visitor.partialSynced()
};
const DataStatus = {
  Synced: _Synced,
  InProgress: _InProgress,
  NotSupported: _NotSupported,
  NotSynced: _NotSynced,
  PartialSynced: _PartialSynced,
  _parse: (value) => {
    switch (value) {
      case "SYNCED": {
        return _Synced;
      }
      case "IN_PROGRESS": {
        return _InProgress;
      }
      case "NOT_SUPPORTED": {
        return _NotSupported;
      }
      case "NOT_SYNCED": {
        return _NotSynced;
      }
      case "PARTIAL_SYNCED": {
        return _PartialSynced;
      }
      default: {
        return {
          value,
          visit: (visitor) => visitor._other(value)
        };
      }
    }
  }
};
