// @fern-api/phyllo/resources/connect/types/PhylloProducts.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var PhylloProducts_exports = {};
__export(PhylloProducts_exports, {
  PhylloProducts: () => PhylloProducts
});
module.exports = __toCommonJS(PhylloProducts_exports);
const _Identity = {
  value: "IDENTITY",
  visit: (visitor) => visitor.identity()
};
const _Engagement = {
  value: "ENGAGEMENT",
  visit: (visitor) => visitor.engagement()
};
const _Income = {
  value: "INCOME",
  visit: (visitor) => visitor.income()
};
const _EngagementAudience = {
  value: "ENGAGEMENT.AUDIENCE",
  visit: (visitor) => visitor.engagementAudience()
};
const _IdentityAudience = {
  value: "IDENTITY.AUDIENCE",
  visit: (visitor) => visitor.identityAudience()
};
const PhylloProducts = {
  Identity: _Identity,
  Engagement: _Engagement,
  Income: _Income,
  EngagementAudience: _EngagementAudience,
  IdentityAudience: _IdentityAudience,
  _parse: (value) => {
    switch (value) {
      case "IDENTITY": {
        return _Identity;
      }
      case "ENGAGEMENT": {
        return _Engagement;
      }
      case "INCOME": {
        return _Income;
      }
      case "ENGAGEMENT.AUDIENCE": {
        return _EngagementAudience;
      }
      case "IDENTITY.AUDIENCE": {
        return _IdentityAudience;
      }
      default: {
        return {
          value,
          visit: (visitor) => visitor._other(value)
        };
      }
    }
  }
};
