// @fern-api/phyllo/resources/connect/types/WorkPlatformCategory.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var WorkPlatformCategory_exports = {};
__export(WorkPlatformCategory_exports, {
  WorkPlatformCategory: () => WorkPlatformCategory
});
module.exports = __toCommonJS(WorkPlatformCategory_exports);
const _Media = {
  value: "MEDIA",
  visit: (visitor) => visitor.media()
};
const _Education = {
  value: "EDUCATION",
  visit: (visitor) => visitor.education()
};
const _Social = {
  value: "SOCIAL",
  visit: (visitor) => visitor.social()
};
const _Entertainment = {
  value: "ENTERTAINMENT",
  visit: (visitor) => visitor.entertainment()
};
const _Lifestyle = {
  value: "LIFESTYLE",
  visit: (visitor) => visitor.lifestyle()
};
const _Business = {
  value: "BUSINESS",
  visit: (visitor) => visitor.business()
};
const WorkPlatformCategory = {
  Media: _Media,
  Education: _Education,
  Social: _Social,
  Entertainment: _Entertainment,
  Lifestyle: _Lifestyle,
  Business: _Business,
  _parse: (value) => {
    switch (value) {
      case "MEDIA": {
        return _Media;
      }
      case "EDUCATION": {
        return _Education;
      }
      case "SOCIAL": {
        return _Social;
      }
      case "ENTERTAINMENT": {
        return _Entertainment;
      }
      case "LIFESTYLE": {
        return _Lifestyle;
      }
      case "BUSINESS": {
        return _Business;
      }
      default: {
        return {
          value,
          visit: (visitor) => visitor._other(value)
        };
      }
    }
  }
};
