// @fern-api/phyllo/core/fetcher/Fetcher.js
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target, mod));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var Fetcher_exports = {};
__export(Fetcher_exports, {
  fetcher: () => fetcher
});
module.exports = __toCommonJS(Fetcher_exports);
var import_axios = __toESM(require("axios"));
const fetcher = async (args) => {
  const headers = {
    "Content-Type": "application/json"
  };
  if (args.headers != null) {
    for (const [key, value] of Object.entries(args.headers)) {
      if (value != null) {
        headers[key] = value;
      }
    }
  }
  try {
    const response = await (0, import_axios.default)({
      url: args.url,
      params: args.queryParameters,
      method: args.method,
      headers,
      data: args.body,
      validateStatus: () => true,
      transformResponse: (response2) => response2,
      timeout: args.timeoutMs ?? 6e4,
      transitional: {
        clarifyTimeoutError: true
      }
    });
    let body;
    if (response.data != null) {
      try {
        body = JSON.parse(response.data) ?? void 0;
      } catch {
        return {
          ok: false,
          error: {
            reason: "non-json",
            statusCode: response.status,
            rawBody: response.data
          }
        };
      }
    }
    if (response.status >= 200 && response.status < 300) {
      return {
        ok: true,
        body
      };
    } else {
      return {
        ok: false,
        error: {
          reason: "status-code",
          statusCode: response.status,
          body
        }
      };
    }
  } catch (error) {
    if (error.code === "ETIMEDOUT") {
      return {
        ok: false,
        error: {
          reason: "timeout"
        }
      };
    }
    return {
      ok: false,
      error: {
        reason: "unknown",
        errorMessage: error.message
      }
    };
  }
};
